app/main/plugins/core/cerebro/settings/Settings/external-settings.js

```diff
@@ -0,0 +1,93 @@
+import React, { PropTypes, Component } from 'react'
+import styles from './styles.css'
+import { Checkbox, Input, Select, Options } from './components'
+
+export default class ExternalSettings extends Component {
+  static propTypes = {
+    settings: PropTypes.object.isRequired,
+    onChange: PropTypes.func.isRequired,
+  }
+
+  changeSetting(plugin, label, value) {
+    this.props.settings[plugin][label].value = value
+    this.props.onChange(this.props.settings)
+  }
+
+  renderSetting(label, setting, plugin) {
+    const { type, description } = setting
+    const value = setting.value || setting.defaultValue
+
+    if (type === 'bool') {
+      return (
+        <Checkbox
+          key={label}
+          value={value}
+          onChange={({ target }) => this.changeSetting(plugin, label, target.checked)}
+          description={description}
+        />
+      )
+    }
+
+    if (type === 'array') {
+      return (
+        <Select
+          key={label}
+          label={label}
+          value={value || []}
+          onChange={newValue => this.changeSetting(plugin, label, newValue)}
+          description={description}
+        />
+      )
+    }
+
+    if (type === 'option') {
+      const { multi } = setting
+      const options = setting.options.map(option => ({ label: option, value: option }))
+
+      return (
+        <Options
+          key={label}
+          label={label}
+          value={value}
+          onChange={newValue => {
+            const changedValue = multi ? newValue.map(val => val.value) : newValue.value
+            this.changeSetting(plugin, label, changedValue)
+          }}
+          description={description}
+          options={options}
+          multi={multi}
+        />
+      )
+    }
+
+    return (
+      <Input
+        key={label}
+        label={label}
+        value={value}
+        onChange={({ target }) => this.changeSetting(plugin, label, target.value)}
```

Comment:
Is it possible to change components a bit so all of them have the same API and onChange function always called with new value, instead of Event or value? In this cases we can do something like this:

```js
const components = {
  bool: Checkbox,
  option: Options,
  array: Select,
}

// ...

const { type, value, defaultValue,  ...restProps } = setting

const FormComponent = components[type] || Input

return <FormComponent 
	value={value || defaultValue}
	key={label}
	label={label}
	onChange={newValue => this.changeSetting(plugin, label, newValue)}
	// ↓ this could contain `multi`, `options` or other specific fields
	...restProps
}
```